Gifski/Gifski.swift

```diff
@@ -1,52 +1,48 @@
 import Foundation
 import AVFoundation
 
-final class Gifski {
-	private var progress: Progress!
-	private var observation: NSKeyValueObservation?
+enum Result {
+	case success
+	case error(GifskiConversionError)
+}
 
-	// `progress.fractionCompleted` is KVO-compliant, but we expose this for convenience
-	var onProgress: ((_ progress: Progress) -> Void)?
+enum GifskiConversionError: LocalizedError {
+	case invalidSettings
+	case generateFrameFailed
+	case addFrameFailed(GifskiWrapperError)
+	case endAddingFramesFailed(GifskiWrapperError)
+	case writeFailed(GifskiWrapperError)
 
-	static func convertFile(
-		_ inputUrl: URL,
-		outputUrl: URL,
-		quality: Double = 1,
-		dimensions: CGSize? = nil,
-		frameRate: Int? = nil
-		) -> Gifski {
-		let gifski = Gifski()
-		gifski.convertFile(
-			inputUrl,
-			outputUrl: outputUrl,
-			quality: quality,
-			dimensions: dimensions,
-			frameRate: frameRate
-		)
-		return gifski
+	var errorDescription: String? {
+		switch self {
+		case .invalidSettings:
+			return "Invalid settings"
+		case .generateFrameFailed:
+			return "Failed to generate frame"
+		case .addFrameFailed(let error):
+			return "Failed to add frame, with underlying error: \(error.localizedDescription)"
+		case .endAddingFramesFailed(let error):
+			return "Failed to end adding frames, with underlying error: \(error.localizedDescription)"
+		case .writeFailed(let error):
+			return "Failed to write to output, with underlying errror: \(error.localizedDescription)"
+		}
 	}
+}
+
+final class Gifski {
 
 	/**
 	- parameters:
 		- frameRate: Clamped to 5...30. Uses the frame rate of `inputUrl` if not specified.
 	*/
-	private func convertFile(
-		_ inputUrl: URL,
-		outputUrl: URL,
-		quality: Double = 1,
+	static func convert(
+		fileAt inputUrl: URL,
+		outputTo outputUrl: URL,
+		withQuality quality: Double = 1,
```

Comment:
I like the Swift convention of human-like parameters, but isn't it inconsistent that it's `withQuality`, but just `dimensions` and `frameRate`?

Reply:
Should maybe skip "with"? I think dropping "with" is a better option than adding it to the last two params.

Reply:
Yes, sounds good.